lib/jellyfish/room_service.ex

```diff
@@ -40,39 +55,63 @@ defmodule Jellyfish.RoomService do
 
   @impl true
   def init(_opts) do
-    :ets.new(:rooms, [:protected, :set, :named_table])
-    {:ok, %{rooms: %{}}}
+    {:ok, %{}}
   end
 
   @impl true
   def handle_call({:create_room, max_peers}, _from, state)
       when is_nil(max_peers) or (is_integer(max_peers) and max_peers >= 0) do
-    {:ok, room_pid} = Room.start_link(max_peers)
+    {:ok, room_pid} = Room.start(max_peers)
     room = Room.get_state(room_pid)
+    Process.monitor(room_pid)
 
     Logger.info("Created room #{inspect(room.id)}")
 
-    :ets.insert(:rooms, {room.id, room_pid})
-
-    {:reply, {:ok, room}, %{state | rooms: Map.put(state.rooms, room.id, room_pid)}}
+    {:reply, {:ok, room}, state}
   end
 
   @impl true
   def handle_call({:create_room, _max_peers}, _from, state),
     do: {:reply, {:error, :bad_arg}, state}
 
   @impl true
-  def handle_call({:delete_room, room_id}, _from, state) when is_map_key(state.rooms, room_id) do
-    state = Map.delete(state, room_id)
-    :ets.delete(:rooms, room_id)
-
-    Logger.info("Deleted room #{inspect(room_id)}")
+  def handle_call({:delete_room, room_id}, _from, state) do
+    response =
+      case find_room(room_id) do
+        {:ok, _pid} ->
+          remove_room(room_id)
+          :ok
+
+        {:error, _} ->
+          {:error, :room_not_found}
+      end
+
+    {:reply, response, state}
+  end
 
-    {:reply, :ok, state}
+  @impl true
+  def handle_info({:DOWN, _ref, :process, _pid, :killed}, state) do
+    {:noreply, state}
   end
 
   @impl true
-  def handle_call({:delete_room, _room_id}, _from, state) do
-    {:reply, {:error, :room_not_found}, state}
+  def handle_info({:DOWN, ref, :process, pid, reason}, state) do
+    Logger.warn("Process (#{inspect(ref)}, #{inspect(pid)}) is down with reason: #{reason}")
+
+    Phoenix.PubSub.broadcast(Jellyfish.PubSub, inspect(pid), :room_crashed)
+
+    {:noreply, state}
+  end
+
+  defp remove_room(room_id) do
+    case find_room(room_id) do
+      {:ok, pid} ->
+        true = Process.exit(pid, :kill)
```

Comment:
Reason `:kill` will shut down the room without any cleanup - I don't think it should be the basic way to stop the room

Reply:
I first tried using `Process.exit(pid,:normal)`, but this led to the strange behavior of the Registry. The registry didn't remove this process, so I decided to fix this by killing the process forcefully. But I didn't know that this would lead to a shutdown process without cleanup so I will try again using normal exit instead of kill.